File: server/src/main/java/com/rrpvm/server/dto/request/ItemCreateDTO.java
package com.rrpvm.server.dto.request;

public class ItemCreateDTO {
    private final String itemName;
    private final String itemHero;
    private final String itemRarity;
    private final String itemDescription;
    private final Double itemPrice;
    private final int itemsAvailable;

    public ItemCreateDTO(String itemName, String itemHero, String itemRarity,String itemDescription, Double itemPrice,int itemAvailable) {
        this.itemName = itemName;
        this.itemHero = itemHero;
        this.itemRarity = itemRarity;
        this.itemDescription = itemDescription;
        this.itemPrice = itemPrice;
        this.itemsAvailable = itemAvailable;
    }

    public String getItemName() {
        return itemName;
    }

    public String getItemHero() {
        return itemHero;
    }

    public String getItemRarity() {
        return itemRarity;
    }

    public Double getItemPrice() {
        return itemPrice;
    }

    public int getItemsAvailable() {
        return itemsAvailable;
    }

    public String getDescription() {
        return itemDescription;
    }
}

File: client/src/model/DTO/request/ItemCreateDTO.ts

//DTO - public access
export default class ItemCreateDTO {
    public itemName:string;
    public itemHero:string;
    public itemRarity:string;
    public itemDescription : string;
    public itemPrice:number;
    public itemsAvailable: number;
    constructor(itemName: string, itemHero: string, itemRarity: string,itemDescription:string, itemPrice: number, itemsAvaliable: number) {
        this.itemName = itemName;
        this.itemHero = itemHero;
        this.itemRarity = itemRarity;
        this.itemDescription = itemDescription;
        this.itemPrice = itemPrice;
        this.itemsAvailable = itemsAvaliable;
    }
};